### PR TITLE
Config

### PR DESCRIPTION
**Pull request by davad**
I added the ElasitcSearch config.

I was looking at the format of URL that Bonsai provides, and it's something like http://user:password@host. Does Mongoosastic know how to handle HTTP authentication like that?

### CODE REVIEW

**Comment by KyleAMathews** [comment]
ElasticSearch by default doesn't support authentication. There are plugins to add HTTP Basic authentication like it looks Bonsai is doing but I'm not any authentication atm for my stuff.

**Comment by davad** [comment]
That's what I figured. 

Does Mongoosastic support HTTP authentication?

**Comment by davad** [comment]
So I think I found the answer to my own question. Mongoosastic uses node-elastical. On https://github.com/ramv/node-elastical/blob/master/lib/client.js, line 87, it looks like I can pass an `:auth` option. 

Edit: Mongoosastic didn't pass authentication info on to node-elastical until version 0.0.12

**Comment by KyleAMathews** [comment]
Sounds perfect.

**Comment by davad** [comment]
Ok, this is actually working now. With the mongoosastic version bump, it passes the auth info on to node-elastical. I have this up and running on heroku using the Bonsai addon as the ElasticSearch provider.

Two documentation notes:
1. Journal is not automatically checking that the index exists before pushing to it, and it doesn't create the index if it doesn't already exist. For Bonsai, I had to create the index using something like `curl -XPOST $ELASTICSEARCH_URL/$ELASTICSEARCH_INDEX`
2. Bonsai expects calls on port 80, but doesn't explicitly include that in $BONSAI_URL. The node-elastical module default to port 9200 if none is explicitly set. I had to set `ELASTICSEARCH_URL=${BONSAI_URL}:80`

File: app_config.coffee [hunk 1]
```diff
@@ -1,2 +1,13 @@
...
+url = require("url")
+# Set these environmental variables to specify a different
+config = mongo_url: process.env.MONGO_URL ?  "mongodb://localhost/journal"
+config.redis_url = process.env.REDIS_URL ? "redis://bogususer:@localhost" #Bogus user to set a blank default password
+config.redis_url = url.parse(config.redis_url)
+
+config.elasticSearchHost = process.env.ELASTICSEARCH_URL ? 'http://localhost'
+config.elasticSearchHost = url.parse(config.elasticSearchHost)
+config.elasticSearchHost.index = process.env.ELASTICSEARCH_INDEX ? 'journal_posts'
+config.elasticSearchHost.protocol = config.elasticSearchHost.protocol.slice(0, -1) #url.parse leaves a colon on the end of the protocol, which node-elastical has a problem with
```
Comment (KyleAMathews) [inline app_config.coffee]: space after the # symbol please

File: app_config.coffee [hunk 2]
```diff
@@ -1,2 +1,13 @@
-module.exports =
-  elasticSearchHost: 'localhost'
+url = require("url")
+# Set these environmental variables to specify a different
```
Comment (KyleAMathews) [inline app_config.coffee]: Comment not finished

**Comment by KyleAMathews** [comment]
Re: documentation note #1. Is creating an index like that a Bonsai specific thing? I've never had to do that with my ElasticSearch server.

**Comment by davad** [comment]
According to the docs, there's an option to [automatically create non-existant indexes](http://www.elasticsearch.org/guide/reference/api/index_/) when pushing to them the first time. I guess it's enabled by default and Bonsai has disabled it?

**Comment by KyleAMathews** [comment]
Perhaps createMapping in mongoosastic create the index?

https://github.com/jamescarr/mongoosastic

**Pull request merged**